packages/mux/src/Mux.ts

```diff
@@ -0,0 +1,132 @@
+import {
+  type TransactionFilter,
+  type OptionsActionParams,
+  OrderType,
+  compressJson,
+} from '@rabbitholegg/questdk'
+import { zeroAddress, type Address } from 'viem'
+import {
+  CHAIN_ID_TO_ORDER_BOOK_ADDRESS,
+  OrderBook__factory,
+  CHAIN_ID_TO_AGGREGATOR_FACTORY_ADDRESS,
+  AggregatorProxyFactory__factory,
+  PositionOrderFlags,
+  AggregatorGmxV2Adapter__factory,
+} from '@mux-network/mux.js'
+import { GNS_ABI } from './abi'
+import { CHAIN_ID_TO_TOKENS, GNS_CONTRACT, GNS_REF_ADDRESS } from './constants'
+import { buildSubAccountIdQuery, chainToWeth } from './helpers'
+import { Chains, buildPathQuery } from './utils'
+
+export const options = async (
+  trade: OptionsActionParams,
+): Promise<TransactionFilter> => {
+  const { chainId, token, amount, recipient, orderType } = trade
+  const muxContract = CHAIN_ID_TO_ORDER_BOOK_ADDRESS[chainId] ?? '0x0'
+  const aggregatorContract =
+    CHAIN_ID_TO_AGGREGATOR_FACTORY_ADDRESS[chainId] ?? '0x0'
+  const subAccountQuery = await buildSubAccountIdQuery(
+    recipient,
+    token,
+    chainId,
+  )
+
+  const tokenIn = token
+    ? token === zeroAddress
+      ? chainToWeth[chainId]
+      : token
+    : undefined
+
+  const getOrderType =
+    orderType === OrderType.Market
+      ? 0
+      : orderType === OrderType.Limit
+      ? { $gt: '0' }
+      : undefined
+
+  const gmxV2OrderFlag =
```

Comment:
I think the double ternary is a little hard to read - wdyt about a helper and maybe a switch statement or an if/else series of returns or something?